docs: update documentation for upstream opentelemetry-php-distro rebase (#373) (#374)
elastic/elastic-otel-php: docs/reference/edot-php/configuration.md

Because the {{edot}} PHP is an extension of the OpenTelemetry PHP SDK, it supports:

* [OpenTelemetry configuration options](#opentelemetry-configuration-options)

> * [EDOT PHP distro configuration options](#edot-php-distro-configuration-options)

Reviewer: ```suggestion
* [EDOT PHP-specific configuration options](#options-only-available-in-edot-php)
```

I would avoid using "distro" as it is ambiguous (could refer to PHP distro and the Elastic distribution as a concept). If used, we would normally go with "distribution/distributions".

Also, sorry to be the bearer of bad news but changing the anchors WILL break some links 😄 Please don't 🙏

EDOT PHP also sets the `OTEL_PHP_AUTOLOAD_ENABLED` option to `true` by default. This turns on automatic instrumentation without requiring any changes to your application code.
Modifying this option will have no effect: EDOT will override it and enforce it as `true`.

> ## EDOT PHP distro configuration options

Reviewer: ```suggestion
## EDOT PHP-specific configuration options
```

> In addition to general OpenTelemetry configuration options, EDOT PHP provides distro-specific configuration options listed below.

Reviewer: ```suggestion
In addition to general OpenTelemetry configuration options, EDOT PHP provides the following distribution-specific configuration options.
```

:::{note}

> **Naming convention change:** Since EDOT PHP has been contributed to the upstream [opentelemetry-php-distro](https://github.com/open-telemetry/opentelemetry-php-distro), all distro-specific options now support **two naming conventions**:

Reviewer: ```suggestion
**Naming convention change:** Since EDOT PHP is now part of the contrib [opentelemetry-php-distro](https://github.com/open-telemetry/opentelemetry-php-distro) project, configuration options support two naming conventions:
```

| Convention | Environment variable prefix | php.ini prefix | Status |
|---|---|---|---|
| **OpenTelemetry (preferred)** | `OTEL_PHP_` | `opentelemetry_distro.` | Recommended |

> | **Elastic (deprecated)** | `ELASTIC_OTEL_` | `elastic_otel.` | Deprecated — still works during the transition period |

Reviewer: ```suggestion
| **Elastic (deprecated)** | `ELASTIC_OTEL_` | `elastic_otel.` | Deprecated — still functional for backward compatibility |
```

Both forms are fully functional. If both are set, the `OTEL_PHP_*` / `opentelemetry_distro.*` value takes precedence.

For example, `ELASTIC_OTEL_LOG_LEVEL` and `OTEL_PHP_LOG_LEVEL` are equivalent, but `OTEL_PHP_LOG_LEVEL` is preferred.
:::

Each option can be set using either an environment variable or the `php.ini` file:

::::{tab-set}

:::{tab-item} Environment variable (preferred)
```bash
export OTEL_PHP_ENABLED=true
```
:::

:::{tab-item} Environment variable (deprecated)
```bash
export ELASTIC_OTEL_ENABLED=true
```
:::

:::{tab-item} php.ini (preferred)
```ini
opentelemetry_distro.enabled=true
```
:::

:::{tab-item} php.ini (deprecated)
```ini
elastic_otel.enabled=true
```
:::

::::

### General configuration

| Option(s)            | Default | Accepted values | Description                                                 |
| -------------------- | ------- | --------------- | ----------------------------------------------------------- |
| OTEL_PHP_ENABLED | `true`    | `true` or `false`   | Enables the automatic bootstrapping of instrumentation code |

> | OTEL_PHP_NATIVE_OTLP_SERIALIZER_ENABLED   | `true`    | `true` or `false`   | Enables the native built-in OTLP Protobuf serializer for maximum performance |

Reviewer: ```suggestion
| OTEL_PHP_ENABLED | `true`    | `true` or `false`   | Activates the automatic bootstrapping of instrumentation code |
| OTEL_PHP_NATIVE_OTLP_SERIALIZER_ENABLED   | `true`    | `true` or `false`   | Activates the native built-in OTLP Protobuf serializer for maximum performance |
```

We don't use "enabled"/"disabled" anymore. There a lot of these in the docs still, and Vale didn't pick them all up. I'll do my best to flag them all.

_Deprecated aliases: `ELASTIC_OTEL_ENABLED`, `ELASTIC_OTEL_NATIVE_OTLP_SERIALIZER_ENABLED`_

### Asynchronous data sending configuration

| Option(s)                                     | Default | Accepted values                                                                                         | Description                                                                                                                          |
| --------------------------------------------- | ------- | ------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------ |

> | OTEL_PHP_ASYNC_TRANSPORT                  | `true`    | `true` or `false`                                                                                           | Use asynchronous (background) transfer of traces, metrics and logs. If `false` - brings back original OpenTelemetry SDK transfer modes |

Reviewer: ```suggestion
| OTEL_PHP_ASYNC_TRANSPORT                  | `true`    | `true` or `false`                                                                                           | Use asynchronous (background) transfer of traces, metrics and logs. If false, reverts to the original OpenTelemetry SDK transfer modes |
```

> | OTEL_PHP_ASYNC_TRANSPORT_SHUTDOWN_TIMEOUT | `30s`     | Integer number with time duration. Set to 0 to disable the timeout. Optional units: ms (default), s, m | Timeout after which the asynchronous (background) transfer will interrupt data transmission during process termination               |

Reviewer: ```suggestion
| OTEL_PHP_ASYNC_TRANSPORT_SHUTDOWN_TIMEOUT | `30s`     | Integer number with time duration. Set to 0 to turn off the timeout. Optional units: ms (default), s, m | Timeout after which the asynchronous (background) transfer will interrupt data transmission during process termination               |
```

| ELASTIC_OTEL_LOG_FILE         |         | Filesystem path                                                                                                                               | Log file name. You can use the %p placeholder where the process ID will appear in the file name, and %t where the timestamp will appear. Please note that the PHP process must have write permissions for the specified path.                                                                                                    |
| ELASTIC_OTEL_LOG_LEVEL_FILE   | `OFF`     | `OFF`, `CRITICAL`, `ERROR`, `WARNING`, `INFO`, `DEBUG`, `TRACE`                                                                                             | Log level for file sink. Set to OFF if you don't want to log to file.                                                                                                                                                                                                                                                            |

> | ELASTIC_OTEL_LOG_LEVEL_STDERR | `OFF`     | `OFF`, `CRITICAL`, `ERROR`, `WARNING`, `INFO`, `DEBUG`, `TRACE`                                                                                             | Log level for the stderr sink. Set to OFF if you don't want to log to a file. This sink is recommended when running the application in a container.                                                                                                                                                                              |

Reviewer: I think it should say "Set to `OFF` to turn off stderr logging." Must be copy-pasted from previous line.

> | OTEL_PHP_LOG_FILE         |         | Filesystem path                                                                                                                               | Log file name. You can use the %p placeholder where the process ID will appear in the file name, and %t where the timestamp will appear. Please note that the PHP process must have write permissions for the specified path.                                                                                                    |

Reviewer: ```suggestion
| OTEL_PHP_LOG_FILE         |         | Filesystem path                                                                                                                               | Log file name. You can use the %p placeholder where the process ID will appear in the file name, and %t where the timestamp will appear. The PHP process must have write permissions for the specified path.                                                                                   |
```

> | OTEL_PHP_LOG_LEVEL_STDERR | `OFF`     | `OFF`, `CRITICAL`, `ERROR`, `WARNING`, `INFO`, `DEBUG`, `TRACE`                                                                                             | Log level for the stderr sink. Set to OFF if you don't want to log to a file. This sink is recommended when running the application in a container.                                                                                                                                                                              |

Reviewer: Sorry, commented on the wrong line. I think it should say "Set to OFF to turn off stderr logging." Must be copy-pasted from previous line.

> | OTEL_PHP_TRANSACTION_URL_GROUPS       |                 | Comma-separated list of wildcard expressions | Allows grouping multiple URL paths using wildcard expressions, such as `/user/*`. For example, `/user/Alice` and `/user/Bob` will be mapped to the transaction name `/user/*`. |

Reviewer: ```suggestion
| OTEL_PHP_TRANSACTION_SPAN_ENABLED     | `true`            | `true` or `false`                                | Activates automatic creation of transaction (root) spans for the webserver SAPI. The name of the span will correspond to the request method and path.                            |
| OTEL_PHP_TRANSACTION_SPAN_ENABLED_CLI | `true`            | `true` or `false`                                | Activates automatic creation of transaction (root) spans for the CLI SAPI. The name of the span will correspond to the script name.                                              |
| OTEL_PHP_TRANSACTION_URL_GROUPS       |                 | Comma-separated list of wildcard expressions | Allows grouping multiple URL paths using wildcard expressions, such as `/user/*`. For example, `/user/Alice` and `/user/Bob` will be mapped to the transaction name `/user/*`. |
```

_Deprecated aliases: `ELASTIC_OTEL_TRANSACTION_SPAN_ENABLED`, `ELASTIC_OTEL_TRANSACTION_SPAN_ENABLED_CLI`, `ELASTIC_OTEL_TRANSACTION_URL_GROUPS`_

### Inferred spans configuration

| Option(s)                                      | Default | Accepted values                                                                                 | Description                                                                                                                                                                                                                                                                                                                                |
| ---------------------------------------------- | ------- | ----------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |

> | OTEL_PHP_INFERRED_SPANS_ENABLED            | `false`   | `true` or `false`                                                                                   | Enables the inferred spans feature.                                                                                                                                                                                                                                                                                                        |

Reviewer: ```suggestion
| OTEL_PHP_INFERRED_SPANS_ENABLED            | `false`   | `true` or `false`                                                                                   | Activates the inferred spans feature.                                                                                                                                                                                                                                                                                                        |
```

| OTEL_PHP_INFERRED_SPANS_REDUCTION_ENABLED  | `true`    | `true` or `false`                                                                                   | If enabled, reduces the number of spans by eliminating preceding frames with the same execution time.                                                                                                                                                                                                                                      |

> | OTEL_PHP_INFERRED_SPANS_STACKTRACE_ENABLED | `true`    | `true` or `false`                                                                                   | If enabled, attaches a stack trace to the span metadata.                                                                                                                                                                                                                                                                                   |

Reviewer: ```suggestion
| OTEL_PHP_INFERRED_SPANS_REDUCTION_ENABLED  | `true`    | `true` or `false`                                                                                   | When active, reduces the number of spans by eliminating preceding frames with the same execution time.                                                                                                                                                                                                                                      |
| OTEL_PHP_INFERRED_SPANS_STACKTRACE_ENABLED | `true`    | `true` or `false`                                                                                   | When active, attaches a stack trace to the span metadata.                                                                                                                                                                                                                                                                                   |
```

| OTEL_PHP_INFERRED_SPANS_SAMPLING_INTERVAL  | 50ms    | Integer number with time duration. Optional units: ms (default), s, m. It can't be set to 0.   | The frequency at which stack traces are gathered within a profiling session. The lower you set it, the more accurate the durations will be. This comes at the expense of higher overhead and more spans for potentially irrelevant operations. The minimal duration of a profiling-inferred span is the same as the value of this setting. |

> | OTEL_PHP_INFERRED_SPANS_MIN_DURATION       | 0       | Integer number with time duration. Optional units: ms (default), s, m. _Disabled if set to 0_. | The minimum duration of an inferred span. Note that the min duration is also implicitly set by the sampling interval. However, increasing the sampling interval also decreases the accuracy of the duration of inferred spans.                                                                                                             |

Reviewer: ```suggestion
| OTEL_PHP_INFERRED_SPANS_MIN_DURATION       | 0       | Integer number with time duration. Optional units: ms (default), s, m. _Deactivated when set to 0_. | The minimum duration of an inferred span. Note that the min duration is also implicitly set by the sampling interval. However, increasing the sampling interval also decreases the accuracy of the duration of inferred spans.                                                                                                             |
```

_Deprecated aliases: `ELASTIC_OTEL_INFERRED_SPANS_ENABLED`, `ELASTIC_OTEL_INFERRED_SPANS_REDUCTION_ENABLED`, `ELASTIC_OTEL_INFERRED_SPANS_STACKTRACE_ENABLED`, `ELASTIC_OTEL_INFERRED_SPANS_SAMPLING_INTERVAL`, `ELASTIC_OTEL_INFERRED_SPANS_MIN_DURATION`_

### Central configuration

The following settings control Central configuration management through OpAMP.

| Option(s)                             | Default                        | Accepted values                                                                               | Description                                                                                                                                                                 |
| ------------------------------------- | ------------------------------ | --------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | OTEL_PHP_OPAMP_ENDPOINT           |              | Valid HTTP or HTTPS URL.                                          | The HTTP or HTTPS endpoint of the OpAMP server. Required to enable Central configuration management. For example, `http://localhost:4320/v1/opamp`. Endpoint must always end with `/v1/opamp`.                            |

Reviewer: ```suggestion
| OTEL_PHP_OPAMP_ENDPOINT           |              | Valid HTTP or HTTPS URL.                                          | The HTTP or HTTPS endpoint of the OpAMP server. Required to activate Central configuration management. For example, `http://localhost:4320/v1/opamp`. Endpoint must always end with `/v1/opamp`.                            |
```

| OTEL_PHP_OPAMP_SEND_TIMEOUT       | 10s                            | Integer number with time duration. Optional units: ms (default), s, m. It can't be set to 0. | Timeout duration for sending messages to the OpAMP server.                                                                                                                   |
| OTEL_PHP_OPAMP_SEND_MAX_RETRIES   | 3                              | Integer ≥ 0                                                                                   | Maximum number of retry attempts for failed message sends.                                                                                                                  |
| OTEL_PHP_OPAMP_SEND_RETRY_DELAY   | 10s                            | Integer number with time duration. Optional units: ms (default), s, m. It can't be set to 0. | Time to wait between retries of failed sends.                                                                                                                                |

> | OTEL_PHP_OPAMP_INSECURE           | false                          | `true` or `false` | If `true`, disables TLS server certificate and hostname verification for the OpAMP HTTPS endpoint (analogous to insecure mode in OTLP exporters). Use ONLY for local testing; leaves the connection vulnerable to MITM. |

Reviewer: ```suggestion
| OTEL_PHP_OPAMP_INSECURE           | false                          | `true` or `false` | If `true`, turns off TLS server certificate and hostname verification for the OpAMP HTTPS endpoint (analogous to insecure mode in OTLP exporters). Use ONLY for local testing; leaves the connection vulnerable to MITM. |
```

> | OTEL_PHP_OPAMP_CERTIFICATE        |                                | Filesystem path (PEM bundle) | Filesystem path to a PEM-encoded CA certificate file or bundle. Must be a readable file; used to verify the OpAMP server when TLS verification is enabled. (same intent as custom root certificates for OTLP). |

Reviewer: ```suggestion
| OTEL_PHP_OPAMP_CERTIFICATE        |                                | Filesystem path (PEM bundle) | Filesystem path to a PEM-encoded CA certificate file or bundle. Must be a readable file; used to verify the OpAMP server when TLS verification is turned on (same intent as custom root certificates for OTLP). |
```
